feeds/wlan-ap/opensync/patches/17-dpp-sta-events.patch

```diff
@@ -0,0 +1,1489 @@
+ extern int nl80211_scan_abort(struct nl_call_param *nl_call_param);
+ extern int nl80211_scan_dump(struct nl_call_param *nl_call_param);
++// real time events handler used by sm_events_report
++extern int nl80211_rt_events_handler(struct nl_call_param *nl_call_param);
++
+ 
+ #endif
+Index: opensync-2.0.5.0/platform/openwrt/src/lib/target/src/radio_nl80211.c
+===================================================================
+--- opensync-2.0.5.0.orig/platform/openwrt/src/lib/target/src/radio_nl80211.c
++++ opensync-2.0.5.0/platform/openwrt/src/lib/target/src/radio_nl80211.c
+@@ -12,6 +12,7 @@
+ #include <net/if.h>
+ 
+ #include <sys/types.h>
++#include <linux/types.h>
+ 
+ #include <linux/sockios.h>
+ #include <linux/nl80211.h>
+@@ -30,6 +31,7 @@
+ #include <libubox/avl.h>
+ #include <libubox/vlist.h>
+ #include <unl.h>
++//#include <bm_ieee80211.h>
+ 
+ #include "target.h"
+ #include "nl80211.h"
+@@ -40,6 +42,7 @@
+ extern struct ev_loop *wifihal_evloop;
+ static struct unl unl;
+ static ev_io unl_io;
++struct rt_events g_rt_events;
+ 
+ static int avl_addrcmp(const void *k1, const void *k2, void *ptr)
+ {
+@@ -384,6 +387,107 @@ static void nl80211_del_phy(struct nlatt
+ 	free(phy);
+ }
+ 
++
++static void mlme_ev_auth(struct nl_msg *msg, struct nlattr **tb)
++{
++    // check ssid and mac
++    struct nlattr *ssid = tb[NL80211_ATTR_SSID];
++    struct nlattr *mac = tb[NL80211_ATTR_MAC];
++    if(!(ssid && mac)) return;
++
++    // add to list
++    target_event_record_t *event_entry = target_event_record_alloc();
++    event_entry->record.event_type = ET_CLIENT_AUTH;
++    // client_auth event pointer
++    dpp_event_record_auth_t *cap = &event_entry->record.client_auth_event;
++    memset(cap, 0, sizeof(dpp_event_record_auth_t));
++    // set data
++    memcpy(cap->sta_mac, nla_data(mac), ETH_ALEN);
++    memcpy(cap->ssid, nla_data(ssid), nla_len(ssid));
++    cap->session_id = get_timestamp();
++    // add to global list
++    ds_dlist_insert_tail(&g_rt_events.event_list, event_entry);
++}
++
++static void mlme_ev_assoc(struct nl_msg *msg, struct nlattr **tb)
++{
++    // check ssid and mac
++    struct nlattr *ssid = tb[NL80211_ATTR_SSID];
++    struct nlattr *mac = tb[NL80211_ATTR_MAC];
++    if(!(ssid && mac)) return;
++
++    // add to list
++    target_event_record_t *event_entry = target_event_record_alloc();
++    event_entry->record.event_type = ET_CLIENT_ASSOC;
++    // client_auth event pointer
++    dpp_event_record_assoc_t *cap = &event_entry->record.client_assoc_event;
++    memset(cap, 0, sizeof(dpp_event_record_assoc_t));
++    // set data
++    memcpy(cap->sta_mac, nla_data(mac), ETH_ALEN);
++    memcpy(cap->ssid, nla_data(ssid), nla_len(ssid));
++    cap->session_id = get_timestamp();
++    // add to global list
++    ds_dlist_insert_tail(&g_rt_events.event_list, event_entry);
++}
++
++static void mlme_ev_disconnect(struct nl_msg *msg, struct nlattr **tb)
++{
++    // check ssid and mac
```

Comment:
// c++ commnt,
/* c99 comment */
also please follow the code style of the rest of the code in the file. it uses tabs for indenting

Reply:
Also seen in **"13-synthetic_client"** patch.

Reply:
in that case 13-synthetic_client needs to also be fixed

Reply:
> in that case 13-synthetic_client needs to also be fixed
@AmmadRehmatCUS